pkg/pillar/devicenetwork/dns.go

```diff
@@ -39,3 +58,115 @@ func ResolvConfToIfname(resolvConf string) string {
 	}
 	return ""
 }
+
+// ResolveWithSrcIP resolves a domain with a given dns server and source Ip
+func ResolveWithSrcIP(domain string, dnsServerIP net.IP, srcIP net.IP) ([]DNSResponse, error) {
+	var response []DNSResponse
+	sourceUDPAddr := net.UDPAddr{IP: srcIP}
+	dialer := net.Dialer{LocalAddr: &sourceUDPAddr}
+	dnsClient := dns.Client{Dialer: &dialer}
+	msg := dns.Msg{}
+	if domain[len(domain)-1] != '.' {
+		domain = domain + "."
+	}
+	msg.SetQuestion(domain, dns.TypeA)
+	dnsClient.Timeout = time.Duration(dnsTimeout)
+	reply, _, err := dnsClient.Exchange(&msg, net.JoinHostPort(dnsServerIP.String(), "53"))
+	if err != nil {
+		return response, fmt.Errorf("dns exchange failed: %v", err)
+	}
+	for _, answer := range reply.Answer {
+		if aRecord, ok := answer.(*dns.A); ok {
+			response = append(response, DNSResponse{
+				IP:  aRecord.A,
+				TTL: aRecord.Header().Ttl,
+			})
+		}
+	}
+
+	return response, nil
+}
+
+// ResolveWithPortsLambda resolves a domain by using source IPs and dns servers from DeviceNetworkStatus as a resolver func ResolveWithSrcIP can be used
+func ResolveWithPortsLambda(domain string, dns types.DeviceNetworkStatus, resolve func(string, net.IP, net.IP) ([]DNSResponse, error)) ([]DNSResponse, []error) {
```

Comment:
We do not really have coding style documented anywhere (mostly relying on `go fmt`), but personally I try to avoid long lines for better readability (my limit is 90 characters at most, but anything below 100 is fine IMHO).

Reply:
Done; perhaps we should integrate https://github.com/segmentio/golines into our build

Reply:
Looks interesting.
Or maybe just having a Yetus complain for a very long line, not sure if there is an option for that.